R/write_camtrapdp.R

```diff
@@ -58,7 +58,7 @@ write_camtrapdp <- function(x, directory, ...) {
   x$data <- NULL
 
   # Remove NA values in x$taxonomic to avoid validation error in frictionless
-  x$taxonomic <- purrr::map(x$taxonomic, ~ .x[!is.na(.x)])
+  x$taxonomic <- purrr::map(x$taxonomic, remove_na_recursive)
```

Comment:
`clean_list` can be called like so: https://github.com/frictionlessdata/frictionless-r/blob/b7fce08225ba2bd4bf5ae2f5be438f225fecfe84/R/create_schema.R#L80-L84

You would have to change `is.null(x)` by `is.na(x)`